core/platform/auth/firebase_auth_services.py

```diff
@@ -105,27 +102,59 @@ def authenticate_request(request):
             https://openid.net/specs/openid-connect-core-1_0.html#IDToken
 
     Args:
-        request: webapp2.Request. The HTTP request to inspect.
+        auth_header: str. The Authorization header taken from the request.
 
     Returns:
-        AuthClaims|None. Claims of the user who authorized the request, or None
-        if the request could not be authenticated.
+        dict(str: *). The Claims embedded into the Authorization header if
+        valid. Otherwise, returns an empty dict.
     """
-    scheme, _, token = request.headers.get('Authorization', '').partition(' ')
+    scheme, _, token = auth_header.partition(' ')
     if scheme != 'Bearer':
-        return None
-
+        return {}
     try:
         with _acquire_firebase_context():
-            claims = firebase_auth.verify_id_token(token)
+            return firebase_auth.verify_id_token(token)
     except (ValueError, firebase_exceptions.FirebaseError) as e:
         logging.exception(e)
-        return None
+        return {}
+
+
+def create_user_auth_details(user_id, auth_id):
+    """Returns a UserAuthDetails object configured with Firebase properties."""
+    return auth_domain.UserAuthDetails(user_id, firebase_auth_id=auth_id)
 
+
+def get_auth_claims_from_request(request):
+    """Authenticates request and returns claims about it's authorizer, if any.
+
+    Oppia specifically expects the request to have a Subject Identifier for the
+    user (Claim Name: 'sub'), and an optional custom claim for super-admin users
+    (Claim Name: 'role').
+
+    Args:
+        request: webapp2.Request. The HTTP request to authenticate.
+
+    Returns:
+        AuthClaims|None. Claims about the user who authorized the request, or
+        None if the request could not be authenticated.
+    """
+    claims = _verify_id_token(request.headers.get('Authorization', ''))
     auth_id = claims.get('sub', None)
     email = claims.get('email', None)
-    # Auth ID is a required Claim, so return None when it is missing.
-    return None if not auth_id else auth_domain.AuthClaims(auth_id, email)
+    role_is_super_admin = claims.get('role', None) == 'admin'
+    if auth_id:
+        return auth_domain.AuthClaims(auth_id, email, role_is_super_admin)
+    return None
+
+
+def disable_auth_associations(user_id):
```

Comment:
Do we have tests that have a user try to log in after their auth assoc has been deleted? Worth making sure that integration pathway works correctly, I think, but I don't see them.

Reply:
So since the [`UserRecord.disabled` documentation](https://firebase.google.com/docs/reference/admin/node/admin.auth.UserRecord#disabled) sucks, I wasn't sure whether it'd be a good idea to set it to false when marking a user for deletion.

I did a simple search for it in the [FirebaseUI-Web repo](https://github.com/firebase/firebaseui-web/blob/acbac901c25f5c5b3325cf3c4d52af0ec763f7a2/soy/strings.soy#L369-L372) and found that a UserFacing error would be raised, though, so I decided to integrate it in after all.

---

TL;DR: The user will see the message: "The user account has been disabled by an administrator." and not be signed-in (and thus, not have an auth_id in the backend).